core-bundle/src/Framework/ContaoFramework.php

```diff
@@ -114,11 +116,15 @@ public function reset(): void
             return;
         }
 
+        ClassLoader::reset();
```

Comment:
Did we have a misconception here? This reset doesn’t seem to uninitialize the framework which means after the first reset `ClassLoader::scanAndRegister()` will never be called again.

And `System::setContainer()` seems to have the same issue?

Reply:
@aschempp Should we 'uninitialize' the framework as well when calling `ContaoFramework::reset()`? Would sound sensible to me. :slightly_smiling_face:

Reply:
That's not fully possible, because initializing the framework will set some constants. What would you uninitialize?

Reply:
The following are not considered “state” IMO (because set by the container or from `Config`) and should not be reset:

- `$this->requestStack`, `$this->scopeMatcher`, `$this->tokenChecker`, `$this->filesystem`, `$this->projectDir`, `$this->errorLevel`
- `$GLOBALS['TL_HOOKS']`,
- `error_reporting()`,
- `ini_set('date.timezone')`, `date_default_timezone_set()`

Needs to be reset (based on the request or method call):

- `$this->request`, `$this->isFrontend`, `$this->adapterCache`, 
- `self::$initialized`, 
- `$_SESSION` (if set to `LazySessionAccess` or `BE_DATA`, `FE_DATA` keys)
- `$GLOBALS['TL_LANGUAGE']`

> That's not fully possible, because initializing the framework will set some constants.

How about throwing an exception if the framework gets (re)initialized and the existing constants do not match the values that should get set (except `TL_START`)?

Reply:
I don’t know, I think it might not make sense for all globals, but maybe for some.